### ISSUE TITLE
How did you convert to GGML?

### ISSUE THREAD

**Issue by Ph0rk0z** [AlpinDale/pygmalion.cpp#8]
I'd like to convert other GPT-J to GGML but can't find any script for it.

**Comment by AlpinDale** [comment]
You can use [this](https://raw.githubusercontent.com/ggerganov/ggml/master/examples/gpt-j/convert-h5-to-ggml.py) script.

**Comment by Ph0rk0z** [comment]
That won't work for a 4-bit quantized model. It just converts a HF to 16 or 32bit.

**Comment by AlpinDale** [comment]
> That won't work for a 4-bit quantized model. It just converts a HF to 16 or 32bit.

Yes, it won't quantize. The original [ggml code](https://github.com/ggerganov/ggml) has the code for quantizing, with instructions on how to do it. I have the code here as well, but it might be slightly outdated compared to the original. You'll need to compile the `quantize.cpp` file instead of the `pyggy.cpp`.

**Comment by Ph0rk0z** [comment]
Oh.. so no python. I have to find a way to build these then. I have the 4-bit model already. Was hoping there was a GPTQ-to-GGML that didn't need a tokenizer.model

**Comment by Ph0rk0z** [comment]
I built it on linux and it quantized gpt4chan. I'd upload the binary but it won't let me attach it. What is the difference between the q4_0 and q4_1?
